llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -15916,21 +15919,30 @@ bool AArch64TargetLowering::optimizeExtendOrTruncateConversion(
 
       DstTy = TruncDstType;
     }
-
-    return createTblShuffleForZExt(ZExt, DstTy, Subtarget->isLittleEndian());
+    IRBuilder<> Builder(ZExt);
+    Value *Result = createTblShuffleForZExt(
```

Comment:
could `createTblShuffleForZExt` still take care of replacing and erasing?

Reply:
I pulled the erasing out of `createTblShuffleForZExt` to avoid the situation down below in the `UIToFPInst` case where a `zext` is created and immediately deleted.

Reply:
Personally I'd go with simpler code, rather than avoiding creating a new ZExt which then gets removed, but don;t feel too strongly about it